Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -208,22 +208,22 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
         return;
     }
 
-    auto& url = document.url();
-    auto& cookieJar = page->cookieJar();
-    auto completionHandler = [promise = WTFMove(promise)] (bool setSuccessfully) {
-        if (!setSuccessfully)
-            promise->reject(TypeError);
-        else
-            promise->resolve();
-    };
-
     Cookie cookie;
     cookie.name = WTFMove(options.name);
     cookie.value = WTFMove(options.value);
     cookie.domain = options.domain.isNull() ? document.domain() : WTFMove(options.domain);
-    cookie.path = WTFMove(options.path);
     cookie.created = WallTime::now().secondsSinceEpoch().milliseconds();
 
+    cookie.path = WTFMove(options.path);
+    if (!cookie.path.isNull()) {
```

Comment:
I don't think this is needed.

Reply:
I may be wrong about this--but if we don't have this check and the path is null, then `cookie.path.startsWith('/')` would be false and we'll return a `TypeError` when we shouldn't right?

Reply:
Oh you're right.  This is needed.